_posts/2021-03-09-bezier-fitting.md

```diff
@@ -0,0 +1,196 @@
+---
+layout: post
+title:  "Fitting cubic Bézier curves"
+date:   2021-03-09 16:10:42 -0700
+categories: [curves]
+---
+<!-- I should figure out a cleaner way to do this include, rather than cutting and pasting. Ah well.-->
+<script type="text/x-mathjax-config">
+	MathJax.Hub.Config({
+		tex2jax: {
+			inlineMath: [['$', '$']]
+		}
+	});
+</script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
+
+Cubic beziers are by far the most common curve representation, used both for design and rendering. A fundamental problem is *curve fitting,* or determining the cubic bezier that's closest to some source curve. Applications include simplifying existing paths, efficiently representing the [parallel curve], and rendering other spline representations such as Euler spiral or [hyperbezier]. A specific feature where curve fitting will be used in [Runebender] is deleting a smooth on-curve point. The intent is to merge two beziers into one, which is another way of saying to find a bezier which most closely approximates the curve formed of the two existing bezier segments.
+
+In spite of the importance of the curve-fitting problem and literature going back more than 30 years, there has been until now no fully satisfactory solution. Existing approaches either miss the optimum solution, are very slow (unsuitable for interactive use), or both.
+
+The main reason the problem is so hard is a basic but little-known fact about cubic beziers: C-shaped beziers come in triplets of very similar shape but quite different parameters. As a result, in general when fitting some arbitrary source curve, there are three local minima. Approaches based on iterative approximation (in addition to being slow) are likely to find just one, potentially missing a closer fit of one of the others.
+
+## How close are two curves?
+
+Before getting into the solution, we'll need to state the problem more carefully. The goal is a cubic bezier that minimizes some error metric with respect to the source curve. But how do we measure the distance between two curves?
+
+One meaningful metric is the [Fréchet distance]. It can be explained with a little story. An aircraft engineer goes on a walk with their dog Pierre, and the path followed by each can be described by a curve. What length leash is required so that the dog is always within that distance of the engineer? That is the Fréchet distance.
```

Comment:
```suggestion
One meaningful metric is the [Fréchet distance]. It can be explained with a little story. An aircraft engineer goes on a walk with their dog Pierre, and the path followed by each can be described by a curve. What is the minimum length leash that allows both to complete their paths? That is the Fréchet distance.
```